pkg/component/component.go

```diff
@@ -402,6 +339,215 @@ func getS2IPaths(podEnvs []corev1.EnvVar) []string {
 	return retVal
 }
 
+// CreateComponent creates component as per the passed component settings
+//	Parameters:
+//		client: occlient instance
+//		componentConfig: the component configuration that holds all details of component
+//		context: the component context indicating the location of component config and hence its source as well
+//		stdout: io.Writer instance to write output to
+//	Returns:
+//		err: errors if any
+func CreateComponent(client *occlient.Client, componentConfig config.LocalConfigInfo, context string, stdout io.Writer) (err error) {
+
+	cmpName := componentConfig.GetName()
+	cmpType := componentConfig.GetType()
+	cmpSrcType := componentConfig.GetSourceType()
+	cmpPorts := componentConfig.GetPorts()
+	cmpMinCPU := componentConfig.GetMinCPU()
+	cmpMaxCPU := componentConfig.GetMaxCPU()
+	cmpMinMemory := componentConfig.GetMinMemory()
+	cmpMaxMemory := componentConfig.GetMaxMemory()
+	cmpSrcRef := componentConfig.GetRef()
+	appName := componentConfig.GetApplication()
+
+	log.Successf("Initializing '%s' component", cmpName)
+	createArgs := occlient.CreateArgs{
+		Name:            cmpName,
+		ImageName:       cmpType,
+		ApplicationName: appName,
+	}
+	createArgs.SourceType = cmpSrcType
+	createArgs.SourcePath = componentConfig.GetSourceLocation()
+
+	if len(cmpPorts) > 0 {
+		createArgs.Ports = cmpPorts
+	}
+
+	if cmpMaxCPU != "" && cmpMinCPU != "" {
+		createArgs.Resources = append(
+			createArgs.Resources,
+			*(util.FetchResourceQuantity(corev1.ResourceCPU, cmpMinCPU, cmpMaxCPU, "")),
+		)
+	}
+	if cmpMaxMemory != "" && cmpMinMemory != "" {
+		createArgs.Resources = append(
+			createArgs.Resources,
+			*(util.FetchResourceQuantity(corev1.ResourceMemory, cmpMinMemory, cmpMaxMemory, "")),
+		)
+	}
+	switch cmpSrcType {
+	case config.GIT:
+		// Use Git
+		if cmpSrcRef != "" {
+			createArgs.SourceRef = cmpSrcRef
+		}
+		if err = CreateFromGit(
+			client,
+			createArgs,
+		); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+		// Git is the only one using BuildConfig since we need to retrieve the git
+		if err = Build(client, createArgs.Name, createArgs.ApplicationName, true, stdout); err != nil {
+			return errors.Wrapf(err, "failed to build component with args %+v", componentConfig)
+		}
+	case config.LOCAL:
+		fileInfo, err := os.Stat(createArgs.SourcePath)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get info of path %+v of component %+v", createArgs.SourcePath, createArgs)
+		}
+		if !fileInfo.IsDir() {
+			return fmt.Errorf("component creation with args %+v as path needs to be a directory", createArgs)
+		}
+		// Create
+		if err = CreateFromPath(client, createArgs); err != nil {
+			return errors.Wrapf(err, "failed to create component with args %+v", createArgs)
+		}
+	case config.BINARY:
+		if err = CreateFromPath(client, createArgs); err != nil {
```

Comment:
Shouldn't we have some path validation here like in case of local? For a non existing path
```
[mrinaldas@localhost odo]$ ./odo create wildfly --binary /home/jljda
 ✓  Checking component
 ✓  Checking component version
Please use `odo push` command to create the component with source deployed
```
But I expect the above to fail

Reply:
That is kind of tricky. It doesn't have to exist when component get created.
Check for existing binary should be there before push is triggered.

Reply:
OK then resolving